Source/Core/Core/Config/GraphicsSettings.cpp

```diff
@@ -4,13 +4,48 @@
 
 #include "Core/Config/GraphicsSettings.h"
 
+#include <optional>
 #include <string>
 
+#include "Common/StringUtil.h"
 #include "Core/Config/Config.h"
 #include "VideoCommon/VideoConfig.h"
 
 namespace Config
 {
+std::optional<int> ConvertFromLegacyEFBScale(int efb_scale)
+{
+  // In game INIs, -1 was used as a special value meaning
+  // "use the value from the base layer but round it to an integer scale".
+  // We only support integer scales nowadays, so we can simply ignore -1
+  // in game INIs in order to automatically use a previous layer's value.
+  if (efb_scale < 0)
+    return {};
+
+  return efb_scale - (efb_scale > 0) - (efb_scale > 2) - (efb_scale > 4);
+}
+
+std::optional<int> ConvertFromLegacyEFBScale(const std::string& efb_scale)
+{
+  int efb_scale_int;
+  if (!TryParse(efb_scale, &efb_scale_int))
+    return {};
+  return ConvertFromLegacyEFBScale(efb_scale_int);
+}
+
+int ConvertToLegacyEFBScale(int efb_scale)
+{
+  return efb_scale + (efb_scale >= 0) + (efb_scale >= 1) + (efb_scale >= 2);
```

Comment:
I just realized that this is wrong. It's supposed to be `efb_scale + (efb_scale >= 0) + (efb_scale > 1) + (efb_scale > 2)`